src/main/java/gregtech/integration/jei/recipe/primitive/MaterialTree.java

```diff
@@ -0,0 +1,124 @@
+import com.google.common.collect.ImmutableList;
+import gregtech.api.unification.OreDictUnifier;
+import gregtech.api.unification.material.type.DustMaterial;
+import gregtech.api.unification.material.type.IngotMaterial;
+import gregtech.api.unification.ore.OrePrefix;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class MaterialTree implements IRecipeWrapper {
+	private final static ImmutableList<OrePrefix> PREFIXES = ImmutableList.of(
+            OrePrefix.dust,
+            OrePrefix.dustSmall,
+            OrePrefix.dustTiny,
+            OrePrefix.ingotHot,
+            OrePrefix.ingot,
+            OrePrefix.gem,
+            OrePrefix.block,
+            OrePrefix.stick,
+            OrePrefix.stickLong,
+            OrePrefix.spring,
+            OrePrefix.gemFlawless,
+            OrePrefix.bolt,
+            OrePrefix.screw,
+            OrePrefix.gemExquisite,
+            OrePrefix.ring,
+            OrePrefix.gear,
+            OrePrefix.frameGt,
+            OrePrefix.nugget,
+            OrePrefix.round,
+            OrePrefix.plate,
+            OrePrefix.plateDense,
+            OrePrefix.gemChipped,
+            OrePrefix.gearSmall,
+            OrePrefix.gemFlawed,
+            OrePrefix.foil,
+            OrePrefix.lens,
+            OrePrefix.wireFine,
+            OrePrefix.wireGtSingle,
+            OrePrefix.cableGtSingle
+	);
+
+	private final List<List<ItemStack>> itemInputs = new ArrayList<>();
+	private final List<List<FluidStack>> fluidInputs = new ArrayList<>();
+
+	private final String name;
+	private final String formula;
+	private final String blastTemp;
+	private final long avgM;
+	private final long avgP;
+	private final long avgN;
+
+	public MaterialTree(DustMaterial material) {
+	    // adding an empty list to itemInputs/fluidInputs makes checking if a prefix exists much easier
+		List<ItemStack> inputDusts = new ArrayList<>();
+		for (OrePrefix prefix : PREFIXES) {
+			inputDusts.add(OreDictUnifier.get(prefix, material));
+		}
+		for (ItemStack stack : inputDusts) {
+			List<ItemStack> matItemsStack = new ArrayList<>();
+			matItemsStack.add(stack);
+			this.itemInputs.add(matItemsStack);
+		}
+
+		FluidStack matFluid = material.getFluid(1000);
+		List<FluidStack> matFluidsStack = new ArrayList<>();
+		if (matFluid != null) {
+			matFluidsStack.add(matFluid);
+		}
+		this.fluidInputs.add(matFluidsStack);
+
+		name = material.getLocalizedName();
+		formula = material.getChemicalFormula();
+		avgM = material.getAverageMass();
+		avgP = material.getAverageProtons();
+		avgN = material.getAverageNeutrons();
+		// not sure of any other good way to signify not having an ebf temp, but this works fine
+		if (material instanceof IngotMaterial) {
+			blastTemp = String.valueOf(((IngotMaterial) material).blastFurnaceTemperature);
```

Comment:
We should have a zero-check here since IngotMaterials will display "Temperature: 0K" if they are not EBF materials currently